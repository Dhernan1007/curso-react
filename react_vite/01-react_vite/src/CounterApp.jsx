import { useState } from 'react';
import PropTypes from 'prop-types';  
export const CounterApp = ({value}) => {
 
  const [ counter, setCounter] = useState(value)
  const handleAdd = ()=>{
    setCounter(counter + 1)
    //setCounter((c)=> c + 1) --> es lo mismo
  }
  const handleRes = ()=>{
    setCounter(counter - 1)
  }
  const handleMult = ()=>{
    setCounter(counter * 2)
  }

  const handleReset = ()=> setCounter( value );

  return (
    <>
      <h1>Counter App</h1>
      <h2> {counter} </h2>
{/* NOTA: onClick={handleAdd} equivale a onClick={(event)=>handleAdd(event)} 
solo que al ser igual el arg de la funcion flecha y del manejador este se omite y se omite la referencia a la funcion flecha */}
      <button onClick={handleAdd}>
        +1
      </button>
      <button onClick={handleRes}>
        -1
      </button>
      <button onClick={handleMult}>
        x2
      </button>
      {/* el area-label es un tipo de etiqueta que me permite identificar el botón esto me ayudará en los test */}
      <button aria-label='btn-reset' onClick={handleReset}>
        Reset
      </button>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

