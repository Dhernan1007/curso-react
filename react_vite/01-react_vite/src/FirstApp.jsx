import PropTypes from 'prop-types';  

export const FirstApp = ({titulo, mensaje, numero})=>{
    return( 
        <>
    <h1 data-testid="test-title"> { titulo } </h1>
    <p>{mensaje}</p>
    <p>{mensaje}</p>
     <p>{numero}</p>
</>
)}

FirstApp.propTypes = {
    /* si lleva isRequired es porque pido que no sea omitido*/
    titulo: PropTypes.string.isRequired,
    mensaje: PropTypes.string.isRequired,
    numero: PropTypes.number,
}

/*
En caso de no pasar props puedo definir alguna por defecto
*/

FirstApp.defaultProps = {
    //titulo: 'No hay titulo',
    mensaje: 'sin cuerpo asociado',
    numero: 1
}