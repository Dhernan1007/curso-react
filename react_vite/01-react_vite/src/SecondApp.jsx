/* si una variable no se usa para afectar a 
un estado o para alguna funcion interna
entonces se coloca fuera del componente
esto para evitar que se rerrenderice */

const newMessage = {
    message: 'Hola compatriota',
    title: 'Darwin'
};
const SecondApp = () => {
  /*cuando se usan fragment estos no se renderizan
  en cambio un div si, y esto puede afectar
  los estilos m√°s adelante*/
    return (
    <>
      <code>{JSON.stringify( newMessage )}</code>
    </>
  )
}
/*
React solo me permite renderizar un objeto si antes
aplico JSON.stringify y paso como argumento el obj
que quiero renderizar
*/
export default SecondApp
