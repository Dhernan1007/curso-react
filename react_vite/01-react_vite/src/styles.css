html, body {
    background-color: #21232A;
    color: #fff;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
    justify-content: space-between;
}

input {
    height: 45px;
    border: solid rgb(28, 200, 28);
    border-radius: 15px;
    font-family: fantasy;
    font-size: 22px;
    text-align: center;
}