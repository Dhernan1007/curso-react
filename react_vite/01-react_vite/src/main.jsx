//imr --> importa react
import React from 'react'
import ReactDOM from 'react-dom/client'
//import SecondApp from './SecondApp'
//import { FirstApp } from './FirstApp'
import './styles.css'
import TablaMultiplicar from './tabla_multiplicar/TablaMultiplicar'
//import {CounterApp} from './CounterApp'


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
    {/* para enviar numeros en react se hace con llaves */}
    {/* <FirstApp titulo = "Mi nombre es Maximo" mensaje = "comandante de la 10ma legiÃ³n" numero = {58}/> */}
    {/* <CounterApp value={10}/> */}
      {/* <SecondApp/> */}
      <TablaMultiplicar/>
    </React.StrictMode>
)
