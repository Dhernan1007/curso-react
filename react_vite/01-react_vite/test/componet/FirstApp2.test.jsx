import {describe, expect, test} from '@jest/globals';
import { render, screen } from '@testing-library/react';
import { FirstApp } from '../../src/FirstApp';
FirstApp


 describe('Pruebas en <FirstApp />', () => {
  
    const titulo = 'Hola Darwin'
    const mensaje = 'Soy el subtitulo'
    test('Debe hacer match con el snapshot', ()=>{
        const { container } = render(<FirstApp titulo={ titulo }/>);
        expect( container ).toMatchSnapshot();

    });

    test('Debe de mostrar el mensaje "Hola Darwin"',()=>{

        /*screen.debug() --> para ver el objeto. tiene asociado el body HTML. Ejecutalo y verás. 
        si lo coloco antes del render se mostrará vacio, si lo coloco despues mostrará lo renderizado hasta ahora
        */

        render(<FirstApp titulo={ titulo }/>);
        expect( screen.getByText(titulo) ).toBeTruthy();
    });

    test('Debe de mostrar el titulo en un h1', ()=>{
        render(<FirstApp titulo={ titulo }/>);
        expect( screen.getByRole('heading', { level: 1}).innerHTML ).toContain(titulo);

    });

    test('Debe de mostrar el subtitulo enviado por props ', () => {
      
        render(
            <FirstApp
                titulo={ titulo }
                mensaje={ mensaje }
            />
        );

        expect( screen.getAllByText(mensaje).length ).toBe(2);
    });
    

});
