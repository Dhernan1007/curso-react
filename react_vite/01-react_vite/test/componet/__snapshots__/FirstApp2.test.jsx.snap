// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola Darwin
     
  </h1>
  <p>
    sin cuerpo asociado
  </p>
  <p>
    sin cuerpo asociado
  </p>
  <p>
    1
  </p>
</div>
`;
